backend/app/endpoints/squads.py

```diff
@@ -3,168 +3,235 @@
 from app.models.squad import Squad
 from app.models.user import User
 from app.permissions.squads import SquadPermissions
-from app.repositories.squads import SquadRepository
+from app.repositories import SquadRepository, UserRepository
 from app.schemas.files import FileResponse
-from app.schemas.squads import SquadResponse, SquadUpsertRequest
+from app.schemas.squads import (
+    SquadInvitationRequest,
+    SquadReplyRequest,
+    SquadResponse,
+    SquadUpdateRequest,
+    SquadUpsertRequest,
+)
 from app.schemas.users import UserResponse
 from app.security.auth import get_current_user
 from fastapi import APIRouter, Depends, HTTPException, UploadFile
 
-router = APIRouter()
+router = APIRouter(dependencies=[Depends(get_current_user)])
+
+
+def get_squad(squad_id: int, squad_repository: SquadRepository = Depends()):
+    """
+    Returns the squad of the corresponding squad_id.
+    Does not check for ownership.
+    Raises 404 if squad is not found
+    """
+    squad = squad_repository.get_squad(squad_id)
+
+    if not squad:
```

Comment:
Based on `backend/app/endpoints/auth.py` change here to:

```suggestion
    if squad is None:
```

Or in `backend/app/endpoints/auth.py` change to `if not [PARAM]:`